lib/ast/lint.js

```diff
@@ -4,24 +4,84 @@ var ASTVisitor = require('juttle/lib/compiler/ast-visitor');
 var _ = require('underscore');
 
 var Lints = {
-    lint: function(ast) {
-        this.visit(ast);
+    lint: function(ast, o) {
+        if (!o.nameField) { throw new Error('Missing required option: nameField'); }
+
+        var options = {
+            nameFields: 0,
+            disallowNameField: false,
+            nameField: o.nameField
+        };
+
+        this.visit(ast, options);
+
+        if (options.nameFields > 1) {
+            throw new Error('Name field ' + options.nameField + ' can be present only once');
+        }
+    },
+
+    isNameField: function(node, nameField) {
+        return node.type === 'UnaryExpression' && node.operator === '*' && node.value === nameField;
     },
 
-    // No time on LHS.
-    visitVariable: function(node) {
-        if (node.name === "time") {
+    isTime: function(node) {
+        return node.type === 'UnaryExpression' && node.operator === '*' && node.value === 'time';
+    },
+
+    visitBinaryExpression: function(node, options) {
+        if (this.isNameField(node.left, options.nameField)) {
+            if (node.operator !== '==' && node.operator !== '=~') {
+                throw new Error('Only equality and regular expressions are supported for name field');
+            }
+
+            if (node.right.type !== 'StringLiteral' && node.right.type !== 'RegularExpressionLiteral') {
+                throw new Error('nameField filter must be a string or a regexp');
+            }
+
+            if (node.right.type === 'StringLiteral' || node.right.type === 'RegularExpressionLiteral') {
```

Comment:
Rather than repeating the type check but inverted, it seems like this would be more readable as:

```
          if (node.right.type !== 'StringLiteral' && node.right.type !== 'RegularExpressionLiteral') {
                throw new Error('nameField filter must be a string or a regexp');
          } else {
                if (node.right.value === '') {
                    throw new Error('nameField filter cannot be empty');
                }
            }
        }
```